### PR TITLE
Implement refresh token

### PR DESCRIPTION
**Pull request by OsQu**
Add `setup_refresh_token` method to `Wsapi::Session`, and request for new access token when WSAPI returns 403.

API works as follows:

```ruby
wsapi = Wsapi::Session.new("oauth2_access_token")
wsapi.setup_refresh_token("client_id", "client_secret", "oauth2_refresh_token") do |new_tokens|
  # ... do something with new tokens, update database or something else
end
```

Also as a bonus, document that WSAPI can be used with OAuth2 tokens too.

### CODE REVIEW

**Comment by OsQu** [comment]
Sure thing.

File: README.md [hunk 1]
```diff
@@ -14,7 +14,15 @@ s = Wsapi::Session.new("deadbeefdeadbeef")
 The constructor also accepts the following options:
   * `:workspace_id`, if not given, user's default workspace is used for queries
   * `:version`, WSAPI version, default is `3.0`
+  * `oauth2_refresh_token`, see: (using WSAPI with OAuth2)[https://github.com/flowdock/rally-wsapi#using-wsapi-with-oauth2]
```
Comment (OsQu) [inline README.md]: Sure thing.

File: spec/lib/session_spec.rb [hunk 2]
```diff
@@ -140,6 +141,56 @@
...
+          "id_token"=>"???",
+          "refresh_token"=>"new_refresh_token",
+          "expires_in"=>15552000,
+          "token_type"=>"Bearer",
+          "access_token"=>"new_access_token"
+        })
+      end
+    end
+
+    context "without refresh_token" do
```
Comment (anttipitkanen) [inline spec/lib/session_spec.rb]: Minor stuff: wouldn't 403 be raised even with refresh_token? The method seems to be 401 as response status.

**Comment by OsQu** [comment]
Done.

**Pull request merged**